platform/darwin/src/MGLStyle.mm

```diff
@@ -211,7 +215,11 @@ - (void)insertLayer:(MGLStyleLayer *)layer belowLayer:(MGLStyleLayer *)otherLaye
     }
 
     const mbgl::optional<std::string> belowLayerId{otherLayer.identifier.UTF8String};
-    self.mapView.mbglMap->addLayer(std::unique_ptr<mbgl::style::Layer>(layer.layer), belowLayerId);
+    try {
+        self.mapView.mbglMap->addLayer(std::unique_ptr<mbgl::style::Layer>(layer.layer), belowLayerId);
```

Comment:
Now that #7282 has landed, including the changes in #6097, you’ll need to wrap any call to `-[MGLStyleLayer addToMapView:]` in this try-catch block or alternatively move this try-catch block to every implementation of `-addToMapView:` in the concrete subclasses of MGLStyleLayer.